hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestMetaWithReplicas.java

```diff
@@ -355,4 +362,58 @@ public void testShutdownOfReplicaHolder() throws Exception {
       assertNotEquals(3, i);
     }
   }
+
+  @Test
+  public void testFailedReplicaAssigment() throws InterruptedException, IOException {
```

Comment:
As the main purpose of the change is to allow Master to complete initialisation in case of meta replica failures, shouldn't this test assert that _HMaster.finishActiveMasterInitialization_ completes properly even with the induced replicas assignment failures? I can see we are asserting that replicas assignment fails, which is not what we actually want to safeguard from.

Reply:
Thanks for the feedback @wchevreuil ! I added a few extra checks to show that the partial meta replica assignment had happened and master is still active and running.

Reply:
Have one question about the test case, do not see any exceptions thrown out during assigning meta replica regions, how do we make sure the following code is working.
```
    try {
      metaBootstrap.assignMetaReplicas();
    } catch (HBaseIOException e){
```
  If we can make sure that the test can compile with or without the changes in non-test modules, it will be great as we can verify that without those changes, the test will fail. Seems with current implementation, it is not possible.

Reply:
@BukrosSzabolcs Your explain about how the test works addressed my concerns, thanks.